Services/Dashboard/Block/classes/class.ilDashboardBlockGUI.php

```diff
@@ -0,0 +1,744 @@
+
+        $sortingCommands = [];
+        $sortings = $this->viewSettings->getSelectableSortingModes();
+        $effectiveSorting = $this->viewSettings->getEffectiveSortingMode();
+        foreach ($sortings as $sorting) {
+            $this->ctrl->setParameter($this, 'sorting', $sorting);
+            $sortingCommands[] = [
+                'txt' => $this->lng->txt('dash_sort_by_' . $sorting),
+                'url' => $this->ctrl->getLinkTarget($this, 'changePDItemSorting'),
+                'asyncUrl' => $this->ctrl->getLinkTarget($this, 'changePDItemSorting', '', true),
+                'active' => $sorting === $effectiveSorting,
+            ];
+            $this->ctrl->setParameter($this, 'sorting', null);
+        }
+
+        if (count($sortingCommands) > 1) {
+            $commandGroups[] = $sortingCommands;
+        }
+
+        if ($manage) {
+            return $commandGroups;
+        }
+
+        $presentationCommands = [];
+        $presentations = $this->viewSettings->getSelectablePresentationModes();
+        $effectivePresentation = $this->viewSettings->getEffectivePresentationMode();
+        foreach ($presentations as $presentation) {
+            $this->ctrl->setParameter($this, 'presentation', $presentation);
+            $presentationCommands[] = [
+                'txt' => $this->lng->txt('pd_presentation_mode_' . $presentation),
+                'url' => $this->ctrl->getLinkTarget($this, 'changePDItemPresentation'),
+                'asyncUrl' => $this->ctrl->getLinkTarget($this, 'changePDItemPresentation', '', true),
+                'active' => $presentation === $effectivePresentation,
+            ];
+            $this->ctrl->setParameter($this, 'presentation', null);
+        }
+
+        if (count($presentationCommands) > 1) {
+            $commandGroups[] = $presentationCommands;
+        }
+
+        $commandGroups[] = [
+            [
+                'txt' => $this->viewSettings->isSelectedItemsViewActive() ?
+                    $this->lng->txt('pd_remove_multiple') :
+                    $this->lng->txt('pd_unsubscribe_multiple_memberships'),
+                'url' => $this->ctrl->getLinkTarget($this, 'manage'),
+                'asyncUrl' => null,
+                'active' => false,
+            ]
+        ];
+
+        return $commandGroups;
+    }
+
+    /**
+     * @throws ilException
+     */
+    public function byType(string $a_type): ilObjectListGUI
+    {
+        $class = $this->objDefinition->getClassName($a_type);
+        if (!$class) {
+            throw new ilException(sprintf("Could not find a class for object type: %s", $a_type));
+        }
+
+        $location = $this->objDefinition->getLocation($a_type);
+        if (!$location) {
+            throw new ilException(sprintf("Could not find a class location for object type: %s", $a_type));
+        }
+
+        $full_class = 'ilObj' . $class . 'ListGUI';
+        require_once $location . '/class.' . $full_class . '.php';
```

Comment:
I don't think this is necessary anymore because the composer should take care of autoloading.